include/sys/arc_impl.h

```diff
@@ -176,6 +176,206 @@ typedef struct l1arc_buf_hdr {
 	abd_t			*b_pabd;
 } l1arc_buf_hdr_t;
 
+enum {
+	L2ARC_DEV_HDR_EVICT_FIRST = (1 << 0)	/* mirror of l2ad_first */
+};
+
+/*
+ * Pointer used in persistent L2ARC (for pointing to log blocks & ARC buffers).
+ */
+typedef struct l2arc_log_blkptr {
+	uint64_t	lbp_daddr;	/* device address of log */
+	/*
+	 * lbp_prop is the same format as the blk_prop in blkptr_t:
```

Comment:
Do we depend on this being the same format as blk_prop?  If not we might want to make that explicit (either in the comments or only have the fields that we use).

Reply:
It uses some BLKPROP macros, so I think we depend on this being the same format as blk_prop. I can update the comments so that it is clear which fields we use.

Reply:
The BLKPROP macros are specific to arc_impl.h, and only coincidentally the same as the BP_GET_ macros that are used with blkptr_t:blk_prop.

Reply:
I had the wrong impression. We do not depend on lbp_prop having the same format as blk_prop. I will update the comments.